### ISSUE TITLE
Determine a validation error code for boolean and geometry

### ISSUE THREAD

**Issue by pedrogit** [edwardsmarc/CASFRI#111]
Target attribute of type boolean and geometry must receive an error code of type boolean or geometry when the source value does not fulfill some validation rules. Unlike for other types, it's hard to find a reserved value for those two types.

**Comment by pedrogit** [comment]
Does CASFRI require any target value to be of type boolean?

**Comment by edwardsmarc** [comment]
We haven't come across any boolean attributes yet.

@prvernier are you aware of any?

**Comment by prvernier** [comment]
No. There are some categorical attributes that take 2 values (e.g., "PF" or "PP") but they are not strictly speaking boolean.

**Comment by edwardsmarc** [comment]
Geometry types could return an empty polygon?

**Comment by pedrogit** [comment]
Empty polygons are often the result of intersection operation or others. There would be no way to distinguish empty polygons from errors. It's a bit like 0 in the integer world.

I was thinking about some very specific polygons like other specific integer values for integers and double precisions. Maybe sometimes just NULL is sufficient (like for boolean).

**Comment by pedrogit** [comment]
Asked the question on GIS Stack Overflow fro geometries: 

https://gis.stackexchange.com/questions/331900/what-could-be-a-proper-and-simple-invalid-wkt-wkb-geometry

Possibilities are:

- EMPTY GEOMETRY
- POINT(0 0)
- NULL
- A small square: POLYGON ((0 0, 0 1, 1 1, 0 0))
- A big square coreving all the possible extent.

**Comment by edwardsmarc** [comment]
For the purpose of creating the GEO tables I have used POLYGON EMPTY as the error code which creates an empty polygon when cast to type geometry. We should revise this error code once we resolve this issue.

**Comment by pedrogit** [comment]
NULL was set as default value for translation and validation error codes when the target type is geometry. That means there is no way to distinguish one error type from the others for geometries.
